splice_machine/src/main/java/com/splicemachine/derby/jdbc/SpliceTransactionResourceImpl.java

```diff
@@ -79,17 +80,17 @@ public SpliceTransactionResourceImpl(String url,Properties info) throws SQLExcep
         }
     }
 
-    public boolean marshallTransaction(TxnView txn) throws StandardException, SQLException {
-        return this.marshallTransaction(txn, null);
+    public void marshallTransaction(TxnView txn) throws StandardException, SQLException {
+        this.marshallTransaction(txn, null);
     }
 
-    public boolean marshallTransaction(TxnView txn, ManagedCache<String, Optional<String>> propertyCache) throws StandardException, SQLException {
-        return this.marshallTransaction(txn, propertyCache, null, null, null);
+    public void marshallTransaction(TxnView txn, ManagedCache<String, Optional<String>> propertyCache) throws StandardException, SQLException {
+        this.marshallTransaction(txn, propertyCache, null, null, null);
     }
 
-    public boolean marshallTransaction(TxnView txn, ManagedCache<String, Optional<String>> propertyCache,
+    public void marshallTransaction(TxnView txn, ManagedCache<String, Optional<String>> propertyCache,
                                        TransactionController reuseTC, String localUserName, Integer sessionNumber) throws StandardException, SQLException{
-        boolean updated = false;
+        assert !updated;
```

Comment:
the method sets the `updated` to true and is public, so it is possible that it is called twice from the outside (which was permissible in the previous version)

I think it is better to return immediately and maybe issue a log.WARN("unexpected") or even throw an exception.

I would also add more documentation about the intended usage of this method, and mark the class `SpliceTransactionResourceImpl` as thread-unsafe.

Reply:
Done

Reply:
Great